instrument/src/Instrument/Types.hs

```diff
@@ -213,13 +246,49 @@ upgradeAggregated_v0 a =
   Aggregated_v1
     { aggTS_v1 = aggTS_v0 a,
       aggName_v1 = MetricName (aggName_v0 a),
-      aggGroup_v1 = (aggGroup_v0 a),
-      aggPayload_v1 = (aggPayload_v0 a)
+      aggGroup_v1 = aggGroup_v0 a,
+      aggPayload_v1 = aggPayload_v0 a
     }
 
 instance Serialize Aggregated_v1 where
   get = (to <$> gGet) <|> (upgradeAggregated_v0 <$> Ser.get)
 
+
+upgradeAggregated_v1 :: Aggregated_v1 -> Aggregated_v2
+upgradeAggregated_v1 a =
+  Aggregated_v2
+    { aggTS_v2 = aggTS_v1 a,
+      aggName_v2 = aggName_v1 a,
+      aggPayload_v2 = aggPayload_v1 a,
+      aggDimensions_v2 = Monoid.mempty
+    }
+
+data Aggregated_v2 = Aggregated_v2
+  { -- | Timestamp for this aggregation
+    aggTS_v2 :: Double,
+    -- | Name of the metric
+    aggName_v2 :: MetricName,
+    -- | Calculated stats for the metric
+    aggPayload_v2 :: AggPayload_v0,
```

Comment:
I was under the impression that if you created a new SafeCopy version like you did for AggPayload that you only had to add the migration for that datatype and not for records that contained it as long as they had derived SafeCopy instances because they'll use the SafeCopy instance for AggPayload to upgrade from any old values. If that's not the case then end users of the library if they ever used one of your safecopied types in their own records, they wouldn't get any sort of type error on upgrade and SafeCopy reads would just silently fail.

Reply:
Yes, for safe-copy this is true. On the other hand, we also have poor's man "safe" copy implemented over serialized as well, see instances for Serialized where `safe-copy` does not help: 
```
instance Serialize Aggregated where
   get = (to <$> gGet) <|> (upgradeAggregated_v1 <$> Ser.get)
   get = (to <$> gGet) <|> (upgradeAggregated_v2 <$> Ser.get) <|> (upgradeAggregated_v2 . upgradeAggregated_v1 <$> Ser.get)
```

Alternatively, we can just revert this change and truncate possible counter values to Int. This should not be a problem if they're reported to the backend periodically (once a few seconds), so the real probability for overflow is low. Any opinions?

Reply:
Once every few seconds seems like a fairly narrow margin of error so I probably wouldn't want to do it. The Serialize instances are unfortunate. I kind of wish we didn't have them if it means that we have to revert to the lowest common denominator from SafeCopy. I suppose we should just leave it how you have it because at least then people who are using the Serialize instance directly aren't broken and people who may be deriving SafeCopy using the types in question *will* be able to benefit from safecopy's automatic migration of record fields.